swift/extractor/main.cpp

```diff
@@ -86,6 +86,8 @@ class Observer : public swift::FrontendObserver {
   }
 
   void configuredCompiler(swift::CompilerInstance& instance) override {
+    // remove default consumers to avoid double messaging
+    instance.getDiags().takeConsumers();
```

Comment:
Does it mean we won't see the messages in the tracer log?

Reply:
here's an example from running `swift/ql/test/extractor-tests/errors`:

```
2023-04-18 07:03:35.999366719 ERRO [extractor/compiler] unresolved.swift:3:10 generic parameter 'T' is not used in function signature (SwiftDiagnosticsConsumer.cpp:45)
2023-04-18 07:03:35.999527236 ERRO [extractor/compiler] unresolved.swift:5:1 cannot explicitly specialize a generic function (SwiftDiagnosticsConsumer.cpp:45)
```

Reply:
One of the concerts I have is that now we have one more layer of indirection when debugging things. I don't have enough experience to judge how bad/good this is, but this is something to keep an eye on.

Reply:
as far as I can tell all the relevant info is still there in the tracer build log, and the conciseness (one line per error without the code snippet) actually helps readability, especially once all extractor logs will follow the same format. Typically the error description and the source line are enough to locate the issue. We can reiterate on this though if it proves to be a problem.